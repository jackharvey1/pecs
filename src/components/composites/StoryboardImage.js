import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Caption from '../atomics/Caption';
import Image from '../atomics/Image';
import { MdCancel } from 'react-icons/md';

const InlineFlexDiv = styled.div`
    position: relative;
    display: inline-flex;
    flex-direction: column;
`;

const DeleteButton = styled.div`
    position: absolute;

    display: flex;
    justify-content: center;
    align-items: center;

    --size: 24px;

    font-size: var(--size);

    top: 0;
    right: 25px;

    @media (min-width: 1024px) {
        right: 12px;
    }

    z-index: 0;

    &:after {
        content: '';

        position: absolute;

        width: 20px;
        height: 20px;
        border-radius: 10px;

        background-color: white;

        z-index: -1;
    }
`;

const StoryboardImage = ({ src, alt, caption, index, deleteCard, addToCaption }) => (
    <InlineFlexDiv>
        <DeleteButton>
            <MdCancel onClick={() => deleteCard(index)} />
        </DeleteButton>
        <Image
            src={src}
            alt={alt}
        />
        <Caption
            caption={caption}
            addToCaption={addToCaption}
            index={index}
        />
    </InlineFlexDiv>
);

StoryboardImage.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
    caption: PropTypes.string,
    index: PropTypes.number.isRequired,
    deleteCard: PropTypes.func.isRequired,
    addToCaption: PropTypes.func.isRequired,
};

export default StoryboardImage;

