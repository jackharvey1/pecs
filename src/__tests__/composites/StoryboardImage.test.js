import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import StoryboardImage from '../../components/composites/StoryboardImage';

describe('StoryboardImage component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(<StoryboardImage
            src="http://image.com/image.png"
            alt="image"
            caption="Caption"
            index={0}
            deleteCard={() => null}
            addToCaption={() => null}
        />).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('calls delete when delete button is pressed', () => {
        const mockDeleteCard = jest.fn();
        const wrapper = mount(<StoryboardImage
            src="http://image.com/image.png"
            alt="image"
            caption="Caption"
            index={0}
            deleteCard={mockDeleteCard}
            addToCaption={() => null}
        />);

        const cancelButton = wrapper.find('MdCancel');
        cancelButton.simulate('click');

        expect(mockDeleteCard).toHaveBeenCalled();
    });
});
