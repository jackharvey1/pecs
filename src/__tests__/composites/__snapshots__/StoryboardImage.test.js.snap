// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryboardImage component renders as intended 1`] = `
.c3 {
  width: 125px;
  margin-top: 12px;
  border: none;
  color: #333333;
  font-size: 16px;
  font-family: Helvetica,Arial,sans-serif;
  text-align: center;
  resize: none;
}

.c3:focus {
  outline: none;
}

.c2 {
  box-shadow: 0 2px 8px 0 rgba(0,0,0,.25);
  margin: auto;
  width: 125px;
  height: 125px;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 24px;
  top: 0;
  right: 0;
  z-index: 0;
}

.c1:after {
  content: '';
  position: absolute;
  width: 20px;
  height: 20px;
  border-radius: 10px;
  background-color: white;
  z-index: -1;
}

@media (min-width:1024px) {
  .c3 {
    width: 150px;
  }
}

@media print {
  .c3::-webkit-input-placeholder {
    color: transparent;
  }

  .c3::-moz-placeholder {
    color: transparent;
  }

  .c3:-ms-input-placeholder {
    color: transparent;
  }

  .c3::placeholder {
    color: transparent;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 150px;
    height: 150px;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      className={undefined}
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
  <img
    alt="image"
    className="c2"
    src="http://image.com/image.png"
  />
  <textarea
    className="c3"
    cols="10"
    id="caption"
    onChange={[Function]}
    placeholder="Add caption here"
    rows="3"
    value="Caption"
  />
</div>
`;
